### DISCUSSION TITLE
502 Bad Gateway on new install

### DISCUSSION THREAD

**Discussion by slaraba** [netbox-community/netbox#12765] [Help Wanted!]
I have installed a fresh instance of Netbox 3.5.1 but am getting a 502 Bad Gateway despite the Netbox WSGI service running and active.

**Comment by adparis99** [comment]
I'd start by verifying that you followed all the steps laid out in [part 5 of the installation guide](https://docs.netbox.dev/en/stable/installation/5-http-server/). Assuming nothing there stands out, a couple other questions:

- Try (if you haven't already) navigating to NetBox from the server itself. Do you still get a 502 error?
- Is there anything that differs between your Apache/nginx setup and the installation guide basics?

**Reply by slaraba** [reply]
I get a 502 as well when running a wget -p to the local host, and IP of the box. 
Unfortunately part 5 of the guide doesn't apply to me since I'm running this on a centos 7 box and it's using ubuntu, so I can't exactly follow the guide. 

Nginx service is running, and netbox seems fine as I can access it fine using the development instance.

**Reply by candlerb** [reply]
Almost certainly your netbox service isn't running. What do `systemctl status netbox` and `journalctl -eu netbox` show?

As for wget, which _port_ did you put in the URL?  If it's 8001 and you get connection refused, then that confirms your netbox service isn't running.  (The out-of-box config runs the gunicorn server on port 8001, as opposed to the development server on port 8000)

**Reply by slaraba** [reply]
`● netbox.service - NetBox WSGI Service
   Loaded: loaded (/etc/systemd/system/netbox.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2023-06-02 12:18:06 BST; 3h 6min ago
     Docs: https://docs.netbox.dev/
 Main PID: 1018 (gunicorn)
   CGroup: /system.slice/netbox.service
           ├─1018 /opt/netbox-3.5.1/venv/bin/python3 /opt/netbox/netbox-3.5.1/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox-3.5.1/netbox --config /opt/netbox/netbox-3.5.1/...
           ├─1191 /opt/netbox-3.5.1/venv/bin/python3 /opt/netbox/netbox-3.5.1/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox-3.5.1/netbox --config /opt/netbox/netbox-3.5.1/...
           ├─1216 /opt/netbox-3.5.1/venv/bin/python3 /opt/netbox/netbox-3.5.1/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox-3.5.1/netbox --config /opt/netbox/netbox-3.5.1/...
           ├─1218 /opt/netbox-3.5.1/venv/bin/python3 /opt/netbox/netbox-3.5.1/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox-3.5.1/netbox --config /opt/netbox/netbox-3.5.1/...
           ├─1225 /opt/netbox-3.5.1/venv/bin/python3 /opt/netbox/netbox-3.5.1/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox-3.5.1/netbox --config /opt/netbox/netbox-3.5.1/...
           └─1229 /opt/netbox-3.5.1/venv/bin/python3 /opt/netbox/netbox-3.5.1/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox-3.5.1/netbox --config /opt/netbox/netbox-3.5.1/...

Jun 02 12:18:06 netbox systemd[1]: Started NetBox WSGI Service.
Jun 02 12:18:07 netbox gunicorn[1018]: [2023-06-02 12:18:07 +0100] [1018] [INFO] Starting gunicorn 20.1.0
Jun 02 12:18:07 netbox gunicorn[1018]: [2023-06-02 12:18:07 +0100] [1018] [INFO] Listening at: http://127.0.0.1:8001 (1018)
Jun 02 12:18:07 netbox gunicorn[1018]: [2023-06-02 12:18:07 +0100] [1018] [INFO] Using worker: gthread
Jun 02 12:18:07 netbox gunicorn[1018]: [2023-06-02 12:18:07 +0100] [1191] [INFO] Booting worker with pid: 1191
Jun 02 12:18:07 netbox gunicorn[1018]: [2023-06-02 12:18:07 +0100] [1216] [INFO] Booting worker with pid: 1216
Jun 02 12:18:07 netbox gunicorn[1018]: [2023-06-02 12:18:07 +0100] [1218] [INFO] Booting worker with pid: 1218
Jun 02 12:18:07 netbox gunicorn[1018]: [2023-06-02 12:18:07 +0100] [1225] [INFO] Booting worker with pid: 1225
Jun 02 12:18:07 netbox gunicorn[1018]: [2023-06-02 12:18:07 +0100] [1229] [INFO] Booting worker with pid: 1229`

**Reply by candlerb** [reply]
And the journalctl entry, after you have hit the server with a web request and got a 502 error?

**Reply by candlerb** [reply]
Also, what does `curl -v http://127.0.0.1:8001/` show?
